contracts/rateproducer/src/rateproducer.cpp

```diff
@@ -454,44 +454,42 @@ namespace eoscostarica {
         auto uniq_rating_index = _ratings.get_index<name("uniqrating")>();
         auto existing_rating = uniq_rating_index.find(uniq_rating);
 
-        if( existing_rating != uniq_rating_index.end() ) {
-            
-            //delete rate info
-            auto itr = uniq_rating_index.erase(existing_rating);
-            
-            //update bp stats
-            float bp_transparency = 0;
-            float bp_infrastructure = 0;
-            float bp_trustiness = 0;
-            float bp_community = 0;
-            float bp_development = 0;
-            uint32_t  bp_ratings_cntr = 0;
-            float  bp_average = 0;
+        check( existing_rating != uniq_rating_index.end(), "Rate should exist before to try to remove it" );
```

Comment:
Message should read: Rating does not exist.

Lets use the work ratings instead of rate, also "before to try" is incorrect , i believe you meant "before trying to remove".